src/AndcultureCode.CSharp.Sitefinity.Core/Models/Services/RestResponseResult.cs

```diff
@@ -0,0 +1,32 @@
+﻿using AndcultureCode.CSharp.Core.Interfaces;
+using AndcultureCode.CSharp.Core.Models.Errors;
+using RestSharp;
+using System.Collections.Generic;
+using System.Net;
+
+namespace AndcultureCode.CSharp.Sitefinity.Core.Models.Services
+{
+    public class RestResponseResult<T> : Result<T>
+    {
+        public HttpStatusCode ExpectedStatusCode { get; }
+
+        /// <summary>
+        /// Provides the actual rest response data in it's original format
+        /// </summary>
+        public IRestResponse RestResponse { get; }
+
+        /// <summary>
+        /// Indicates whether the RestResponse's StatusCode was of the expected type after the request was made
+        /// and it's data value was returned
+        /// </summary>
+        public bool WasExpectedStatusCode => ExpectedStatusCode == RestResponse.StatusCode;
+        public bool WasUnexpectedStatusCode => !WasExpectedStatusCode;
```

Comment:
Looping @HeyKos  in on this since he commented on the original PR from within our project.  He mentioned having one or the other `WasExpectedStatusCode` & `WasUnexecptedStatusCode` but not both.  I think having both are valuable.

Reply:
Yea, I think having both is nice just for readability. Until someone decides to write `!WasUnexpectedStatusCode` XD